src/Components/Admin/NewUserDialog.js

```diff
@@ -35,15 +33,15 @@ function NewUserDialog({ onSubmit, close, confirm }) {
     setIsSubmitting(true);
     try {
       const data = { ...formState };
-      data.programs = Array.from(formState.programs).map((program) => ({
-        id: program,
-        role: "reviewer"
-      }));
-
-      const user = await createUserAPI(data);
-      close();
-      onSubmit && onSubmit(user);
-      confirm();
+      const user = await createOrAddUserProgramAPI(programId, {
+        email: data.email,
+        role: data.role
+      }).then(() => {
+        close();
+        window.location.reload();
+        onSubmit && onSubmit(user);
+        confirm();
```

Comment:
I noticed the page wasn't reloading like when a user is edited/deleted, so I copied `window.location.reload()` from the edit user dialog. Let me know if this isn't a best practice!